databricks/koalas/indexes.py

```diff
@@ -690,6 +709,96 @@ def levels(self) -> list:
         # names, so e.g. if ['b', 'd', 'a'] will return as ['a', 'b', 'd']
         return [sorted(col) for col in row]
 
+    def _get_level_number(self, level):
+        """
+        Return the level number if a valid level is given.
+
+        **this is an internal method**
+        """
+        count = self.names.count(level)
+        if (count > 1) and not isinstance(level, int):
+            raise ValueError(
+                "The name %s occurs multiple times, use a level number" % level
+            )
+        try:
+            level = self.names.index(level)
+        except ValueError:
+            if not isinstance(level, int):
+                raise KeyError("Level %s not found" % str(level))
+            elif level < 0:
+                level += self.nlevels
+                if level < 0:
+                    orig_level = level - self.nlevels
+                    raise IndexError(
+                        "Too many levels: Index has only %d "
+                        "levels, %d is not a valid level number"
+                        % (self.nlevels, orig_level)
+                    )
+            # Note: levels are zero-based
+            elif level >= self.nlevels:
+                raise IndexError(
+                    "Too many levels: Index has only %d levels, "
+                    "not %d" % (self.nlevels, level + 1)
+                )
+        return level
+
+    def _get_level_values(self, level):
+        """
+        Return vector of label values for requested level,
+        equal to the length of the index
+
+        **this is an internal method**
+
+        Parameters
+        ----------
+        level : int level
+
+        Returns
+        -------
+        values : Index
+        """
+        lev = self.levels[level]
```

Comment:
We should avoid using `levels` property since it runs a Spark job.

Reply:
Also the returned value will be different.

```py
>>> pd.MultiIndex.from_tuples([('x', 'a'), ('x', 'b'), ('y', 'a')]).get_level_values(0)
Index(['x', 'x', 'y'], dtype='object')
```

Reply:
The comments here are resolved at https://github.com/databricks/koalas/pull/1517/commits/dd8683c1fdf6ab9a2f86c822cf6a5fad3c581c04.